lib/topological_inventory/ansible_tower/collectors_pool.rb

```diff
@@ -29,7 +29,10 @@ def source_valid?(source, secret)
     end
 
     def new_collector(source, secret)
-      TopologicalInventory::AnsibleTower::Collector.new(source.source, source.host, secret["username"], secret["password"], metrics)
+      url = URI::Generic.build(:scheme => source.scheme.to_s.strip.presence || 'https',
+                               :host   => source.host.to_s.strip,
```

Comment:
@slemrmartin what is `.to_s.strip` buying you here?  Are you trying to make sure that it isn't `nil`?  How about the strip are we getting leading or trailing whitespace from the Settings?

Reply:
yes, `.to_s` ensures it isn't `nil`. 

`strip` is there because values are provided by user inputs in UI, parsed, splitted and I maybe some other operations, so I want to be sure.

Reply:
btw I cannot check it if is possible to provide anythiing throught UI, because insights-proxy is not working now due to domain change

Reply:
Okay, nil is fine with URI build but I get if you want to strip the string as well.